Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs

```diff
@@ -1250,41 +1257,159 @@ List<ElementInfo> IWindowExplorer.GetVisibleControls(List<eElementType> filtered
                         if (cattr.Value == "false") continue;
                     }
                 }
-                //AppiumElementInfo AEI = GetElementInfoforXmlNode(nodes[i]);                
-                //list.Add(AEI);
+                ElementInfo EI = await GetElementInfoforXmlNode(nodes[i]);
+                list.Add(EI);
             }
 
             return list;
         }
 
-        //private AppiumElementInfo GetElementInfoforXmlNode(XmlNode xmlNode)
-        //{
-        //    AppiumElementInfo AEI = new AppiumElementInfo();
-        //    AEI.ElementTitle = GetNameFor(xmlNode);            
-        //    AEI.ElementType = GetAttrValue(xmlNode, "class");
-        //    AEI.Value = GetAttrValue(xmlNode, "text");
-        //    if (string.IsNullOrEmpty(AEI.Value)) 
-        //        {
-        //            AEI.Value = GetAttrValue(xmlNode, "content-desc");
-        //        }
-        //    AEI.XmlNode = xmlNode;
-        //    AEI.XPath = GetNodeXPath(xmlNode);
-        //    AEI.WindowExplorer = this;
-
-        //    return AEI;
-        //}
+        private async Task<ElementInfo> GetElementInfoforXmlNode(XmlNode xmlNode)
+        {
+            ElementInfo EI = new ElementInfo();
+            EI.ElementTitle = GetNameFor(xmlNode);
+            Tuple<string, eElementType> Elementype = GetElementTypeEnum(xmlNode);
+            EI.ElementType = Elementype.Item1;           //GetAttrValue(xmlNode, "class");
+            EI.ElementTypeEnum = Elementype.Item2;
+            EI.Value = GetAttrValue(xmlNode, "text");
+            if (string.IsNullOrEmpty(EI.Value))
+            {
+                EI.Value = GetAttrValue(xmlNode, "content-desc");
+            }
+            EI.ElementObject = xmlNode;
+            EI.XPath = await GetNodeXPath(xmlNode);
+            EI.WindowExplorer = this;
+
+            return EI;
+        }
 
-        private string GetNodeXPath(XmlNode xmlNode)
+        public static Tuple<string, eElementType> GetElementTypeEnum(XmlNode xmlNode)
         {
-            string XPath = GetXPathToNode(xmlNode);
-            return XPath;
+            Tuple<string, eElementType> returnTuple;
```

Comment:
is this different from Selenium? can't we reuse the element type been found by selenium when running in Web mode?